### ISSUE TITLE
CatchUnhandledException sets Thread.CurrentThread.Name, which throws InvalidOperationException if the thread name has already been set

### ISSUE THREAD

**Issue by wjrogers** [Topshelf/Topshelf#389]
I have written a Topshelf service that starts several threads and names those threads for logging purposes. When my service throws an unhandled (fatal) exception, Topshelf's `ConsoleRunHost.CatchUnhandledException` method tries to set the current thread's name, which throws `InvalidOperationException` because `Thread.Name` is a write-once property.

Throwing this exception in the unhandled exception handler obscures important information about why the process crashed. Topshelf should not try to set the current thread's name. At the very least, it should check that the thread name has not been set (`Thread.CurrentThread.Name == null`) before it tries to set the thread name.

```
System.InvalidOperationException occurred
  HResult=0x80131509
  Message=This property has already been set and cannot be modified.
  Source=mscorlib
  StackTrace:
   at System.Threading.Thread.set_Name(String value)

mscorlib.dll!System.Threading.Thread.Name.set(string value)
Topshelf.dll!Topshelf.Hosts.ConsoleRunHost.CatchUnhandledException(object sender, System.UnhandledExceptionEventArgs e)
[Native to Managed Transition]
```

**Comment by phatboyg** [comment]
I am closing all issues that haven't seen any activity. You're welcome to submit a PR to resolve the issue or provide more information.

**Comment by jods4** [comment]
@phatboyg please don't close this issue.
It has seen no activity because there's no _"more information"_ to be provided. 
The defect is clear and annoying when it hits you, and its potential resolution is clear as well.

I can understand that the package maintainers don't have time to fix it right now, neither do I. 
But it's worth keeping open until someone does. If only because people will look for it on Google (I did) and open it again.

**Comment by phatboyg** [comment]
This was fixed in 03bde8963da8c057ffc6df07ef9600ffdf6047c1

**Comment by jods4** [comment]
Great news, thank you! 🎉
